demo/src/app/components/typeahead/demos/http/typeahead-http.ts

```diff
@@ -10,25 +10,27 @@ import 'rxjs/add/operator/map';
 import 'rxjs/add/operator/switchMap';
 import 'rxjs/add/operator/merge';
 
+const WIKI_URL = 'https://en.wikipedia.org/w/api.php';
+const PARAMS = new HttpParams({
+  fromObject: {
+    action: 'opensearch',
+    format: 'json',
+    origin: '*'
+  }
+});
+
 @Injectable()
 export class WikipediaService {
-  constructor(private _jsonp: Jsonp) {}
+  constructor(private http: HttpClient) {}
 
   search(term: string) {
     if (term === '') {
       return Observable.of([]);
     }
 
-    let wikiUrl = 'https://en.wikipedia.org/w/api.php';
-    let params = new URLSearchParams();
-    params.set('search', term);
-    params.set('action', 'opensearch');
-    params.set('format', 'json');
-    params.set('callback', 'JSONP_CALLBACK');
-
-    return this._jsonp
-      .get(wikiUrl, {search: params})
-      .map(response => <string[]> response.json()[1]);
+    return this.http
+      .get(WIKI_URL, {params: PARAMS.set('search', term)})
```

Comment:
Looks kind of strange to mutate a shared object.. How about `new HttpParams(Object.assign(PARAMS, {'search': term})`?

Reply:
It's not mutating the existing one, see [implementation and doc](https://github.com/angular/angular/blob/6911a250ef82ef0cf41c42d34b75885a25c0301b/packages/common/http/src/params.ts#L164).